[FIX] #157 Check that the use_parent_address field exists before using i...

ccomb opened the pull request:
...t. Also revert the hardcoded layout bloc to ease merge from upstream.

Fix for OCB issue #157

ccomb committed:
[FIX] #157 Check that the use_parent_address field exists before using it. Also revert the hardcoded layout bloc to ease merge from upstream

petrus-v commented:
:+1:  (review only not tested)

pedrobaeza commented on openerp/addons/base/res/res_partner.py:
```diff
@@ -87,6 +70,9 @@ def fields_view_get_address(self, cr, uid, arch, context={}):
                     tree = etree.fromstring(v)
                     # needs to apply modifiers
                     for childnode in tree.getchildren():
+                        if 'use_parent_address' in self._all_columns:
```
But this is going to apply the modifier to each element found, including div and so on, doesn't it?

petrus-v replied:
I don't think so we are here on children of `adress_format` css class:
``doc.xpath("//div[@class='address_format']")``

ccomb committed:
don't apply the attrs on the <br/>

pedrobaeza commented:
And what about putting the attrs in the div tag?

ccomb commented:
Good idea, I'm trying

ccomb commented:
Nope, that does'nt work on the div.

pedrobaeza commented:
Thanks for the patch.

:+1:

guewen commented:
Are we guaranteed that the field `use_parent_address` exists in the view?

ccomb commented:
Good remark, my check is done on the model, not the view. I'm looking again.

ccomb committed:
Search use_parent_address in the view instead of the model

ccomb committed:
Saved one line

ccomb commented:
@guewen you were right, it should be better now.  Doing the same for v8

guewen commented:
:+1:

ccomb commented:
@pedrobaeza sorry, could you remerge and include the latest commit ? I had removed the check of the type filed in a previous commit.
This does not hurt, but having a `<br/>` tag with a "attrs" is not very useful...

I've just fixed on v8 as well before merge.

pedrobaeza commented:
OK, I will amend the commit in a couple of hours.

pedrobaeza commented:
I have made it in a new commit (https://github.com/OCA/OCB/commit/0ae923b7307ae0f2656fc64711f9e6e2d4d5a1ce) to avoid messing people that have made pulls in between.

ccomb commented:
Thank you!
